File: control-assurance-sp/src/types/NAORecommendation.ts
import { IEntity } from "./Entity";

export interface INAORecommendation extends IEntity {
    RecommendationDetails?: string;
    Conclusion?: string;
    OriginalTargetDate?: string;
    NAOPublicationId?: number;
    NAOUpdateStatusTypeId?: number;
}

export class NAORecommendation implements INAORecommendation {
    public ID: number = 0;
    public Title: string = null;
    public RecommendationDetails?: string = null;
    public Conclusion?: string = null;
    public OriginalTargetDate?: string = null;
    public NAOPublicationId = null;
    public NAOUpdateStatusTypeId = null;

}

File: ControlAssuranceAPI/Models/NAORecommendation.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class NAORecommendation
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public string? RecommendationDetails { get; set; }

    public int? NAOPublicationId { get; set; }

    public int? NAOUpdateStatusTypeId { get; set; }

    public string? Conclusion { get; set; }

    public string? OriginalTargetDate { get; set; }

    public virtual ICollection<NAOAssignment> NAOAssignments { get; } = new List<NAOAssignment>();

    public virtual NAOPublication? NAOPublication { get; set; }

    public virtual NAOUpdateStatusType? NAOUpdateStatusType { get; set; }

    public virtual ICollection<NAOUpdate> NAOUpdates { get; } = new List<NAOUpdate>();
}

public class NAORecommendationView_Result
{
    public int ID { get; set; }
    public string? Title { get; set; }
    public string? RecommendationDetails { get; set; }
    public string? TargetDate { get; set; }
    public string? RecStatus { get; set; }
    public string? AssignedTo { get; set; }
    public string? AssignedToIds { get; set; }
    public string? UpdateStatus { get; set; }
}